django/db/backends/sqlite3/introspection.py

```diff
@@ -192,7 +192,7 @@ def get_primary_key_column(self, cursor, table_name):
         fields_sql = create_sql[create_sql.index('(') + 1:create_sql.rindex(')')]
         for field_desc in fields_sql.split(','):
             field_desc = field_desc.strip()
-            m = re.match('(?:"(.*)"|([\w-]+)).*PRIMARY KEY.*', field_desc)
+            m = re.match('(?:(?:"|\[|`)(.*)(?:"|\]|`)|([\w-]+)).*PRIMARY KEY.*', field_desc)
```

Comment:
A non-capture group that encapsulates two additional non-capture groups to help define the first capture group's expression for quotation matching. If the first capture group is empty, we know a second capture group (unquoted column) is present.

Reply:
With your suggestion, ```(?:(?:["`\[])(.*)(?:["`\]])|([\w-]+)).*PRIMARY KEY.*```. I think that's subjective at this point. If you would still like, I will commit this change.

Reply:
You need to have the non-capture groups, otherwise you're only going to match ```id" int``` in your first capture group, following your example.